### ISSUE TITLE
Cant find Midea applience via cloud. Error code: 3004 message: value is illegal

### ISSUE THREAD

**Issue by Enton-72** [nbogojevic/midea-beautiful-air#25]
## Version of the library
Library version=0.9.5

### Environment
LXD Host: Debian 11 (6.0.0-0.deb11.6-amd64)
LXC Container: Debian 12 (6.0.0-0.deb11.6-amd64)
Network: Full access
Python: 3.11.2

## Model of appliance where problem occurs
Model: MSMBBU-12HRFN1-QRD0G

## Describe the bug
The bug has occurred when I was trying to obtain TOKEN/KEY for our AC appliances via username/password.
Command:
```
midea-beautiful-air-cli --log DEBUG --verbose discover --app "MSmartHome" --account 'mail@example.com' --password 'password' --credentials
```
Local discover is working (without "--account" and "--password" arguments).

I have tried to do discovery using msmart lib (https://github.com/mac-zhou/midea-msmart) and got the same error but with more info:
```ValueError: (1006, 'mobile=main@example.com value is illegal.')```

Tried with china server/proxied and phone number/email.

## Logs
```
Log:
2023-08-02 08:35:37 mideacontrol midea_beautiful[6923] DEBUG Library version=0.9.15
2023-08-02 08:35:37 mideacontrol midea_beautiful.cloud[6923] DEBUG HTTP request /v1/multicloud/platform/user/route: {'x-recipe-app': '1010', 'Authorization': 'Basic ********************************************************', 'sign': 'f2c73eee6c6d166bbff82a417f47acc4e1cc285c3c62e8a130762cb7817fda74', 'secretVersion': '1', 'random': '1690965337', 'version': '2.22.0', 'systemVersion': '8.1.0', 'platform': '0', 'Accept-Encoding': 'identity', 'Content-Type': 'application/json'} {'appId': 1010, 'format': 2, 'clientType': 1, 'language': 'en_US', 'src': 1010, 'stamp': '20230802083537', 'userName': 'mail@example.com', 'appVNum': '2.22.0', 'appVersion': '2.22.0', 'clientVersion': '2.22.0', 'platformId': '1', 'reqId': 'e0e0e38a70ea17bdbe9a755e4047b72c', 'retryCount': '3', 'uid': '', 'userType': '0'}
2023-08-02 08:35:37 mideacontrol urllib3.connectionpool[6923] DEBUG Starting new HTTPS connection (1): mp-prod.appsmb.com:443
2023-08-02 08:35:38 mideacontrol urllib3.connectionpool[6923] DEBUG https://mp-prod.appsmb.com:443 "POST /mas/v5/app/proxy?alias=/v1/multicloud/platform/user/route HTTP/1.1" 200 168
2023-08-02 08:35:38 mideacontrol midea_beautiful.cloud[6923] DEBUG HTTP response text: {"msg":"ok","code":0,"data":{"masUrl":"https://mp-prod.appsmb.com/mas/v5/app/proxy?alias=","sseUrl":"https://sse.msmartlife.net","idAdapt":"131329","countryCode":"HK"}}
2023-08-02 08:35:38 mideacontrol midea_beautiful.cloud[6923] DEBUG HTTP response: {'msg': 'ok', 'code': 0, 'data': {'masUrl': 'https://mp-prod.appsmb.com/mas/v5/app/proxy?alias=', 'sseUrl': 'https://sse.msmartlife.net', 'idAdapt': '131329', 'countryCode': 'HK'}}
2023-08-02 08:35:38 mideacontrol midea_beautiful.cloud[6923] DEBUG using key='data', result={'masUrl': 'https://mp-prod.appsmb.com/mas/v5/app/proxy?alias=', 'sseUrl': 'https://sse.msmartlife.net', 'idAdapt': '131329', 'countryCode': 'HK'}
2023-08-02 08:35:38 mideacontrol midea_beautiful.cloud[6923] DEBUG HTTP request /v1/user/login/id/get: {'x-recipe-app': '1010', 'Authorization': 'Basic ********************************************************', 'sign': '63a766c37601fd23ed6125cc26d7fbfb142dbd575ca61bfc15fc51717d4c7bdc', 'secretVersion': '1', 'random': '1690965338', 'version': '2.22.0', 'systemVersion': '8.1.0', 'platform': '0', 'Accept-Encoding': 'identity', 'Content-Type': 'application/json'} {'appId': 1010, 'format': 2, 'clientType': 1, 'language': 'en_US', 'src': 1010, 'stamp': '20230802083538', 'loginAccount': 'mail@example.com', 'appVNum': '2.22.0', 'appVersion': '2.22.0', 'clientVersion': '2.22.0', 'platformId': '1', 'reqId': 'e603d06e5917e4c5e0e7ff27a4103a54', 'retryCount': '3', 'uid': '', 'userType': '0'}
2023-08-02 08:35:38 mideacontrol urllib3.connectionpool[6923] DEBUG Starting new HTTPS connection (1): mp-prod.appsmb.com:443
2023-08-02 08:35:38 mideacontrol urllib3.connectionpool[6923] DEBUG https://mp-prod.appsmb.com:443 "POST /mas/v5/app/proxy?alias=/v1/user/login/id/get HTTP/1.1" 200 41
2023-08-02 08:35:38 mideacontrol midea_beautiful.cloud[6923] DEBUG HTTP response text: {"code":"3004","msg":"value is illegal."}
2023-08-02 08:35:38 mideacontrol midea_beautiful.cloud[6923] DEBUG HTTP response: {'code': '3004', 'msg': 'value is illegal.'}
2023-08-02 08:35:38 mideacontrol midea_beautiful.cloud[6923] DEBUG Restarting session: '3004' - 'value is illegal.'

```
If you need any additional info I will try my best to provide it.

Thank you.

**Comment by dkerr64** [comment]
Camping on that I just bought a Midea dehumidifier and have run into exactly this problem.  I know the device is registered to the network properly as I can discover it using IP address...
```
root@homebridge:~# midea-beautiful-air-cli discover --address 192.168.29.131
id 000000P0000000Q1B88C291B363A0000/150633093209019
  id      = 150633093209019
  addr    = 192.168.29.131
  s/n     = 000000P0000000Q1B88C291B363A0000
  model   = Dehumidifier
  ssid    = net_a1_363A
  online  = True
  name    = 150633093209019
  running = False
  humid%  = 45
  target% = 50
  temp    = 0
  fan     = 40
  tank    = False
  mode    = 0
  ion     = False
  filter  = False
  pump    = False
  defrost = False
  sleep   = False
  error   = 0
  supports= {}
  version = 3
root@homebridge:~#
```
But as soon as I try and retrieve credentials using my account/password then I get the 3004 value is illegal error.

```
root@homebridge:~# midea-beautiful-air-cli --verbose --log DEBUG --no-redact discover --address 192.168.29.131 --account David@<redacted> --password <redacted> --credentials --app MSmartHome
DEBUG:midea_beautiful:Library version=0.9.15
DEBUG:midea_beautiful.cloud:HTTP request /v1/multicloud/platform/user/route: {'x-recipe-app': '1010', 'Authorization': 'Basic <redacted>=', 'sign': '<redacted>', 'secretVersion': '1', 'random': '1691159233', 'version': '2.22.0', 'systemVersion': '8.1.0', 'platform': '0', 'Accept-Encoding': 'identity', 'Content-Type': 'application/json'} {'appId': 1010, 'format': 2, 'clientType': 1, 'language': 'en_US', 'src': 1010, 'stamp': '20230804102713', 'userName': 'david@<redacted>', 'appVNum': '2.22.0', 'appVersion': '2.22.0', 'clientVersion': '2.22.0', 'platformId': '1', 'reqId': '5a87a707a7dd0efeb733536e624116ed', 'retryCount': '3', 'uid': '', 'userType': '0'}
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): mp-prod.appsmb.com:443
DEBUG:urllib3.connectionpool:https://mp-prod.appsmb.com:443 "POST /mas/v5/app/proxy?alias=/v1/multicloud/platform/user/route HTTP/1.1" 200 168
DEBUG:midea_beautiful.cloud:HTTP response text: {"msg":"ok","code":0,"data":{"masUrl":"https://mp-prod.appsmb.com/mas/v5/app/proxy?alias=","sseUrl":"https://sse.msmartlife.net","idAdapt":"131329","countryCode":"HK"}}
DEBUG:midea_beautiful.cloud:HTTP response: {'msg': 'ok', 'code': 0, 'data': {'masUrl': 'https://mp-prod.appsmb.com/mas/v5/app/proxy?alias=', 'sseUrl': 'https://sse.msmartlife.net', 'idAdapt': '131329', 'countryCode': 'HK'}}
DEBUG:midea_beautiful.cloud:using key='data', result={'masUrl': 'https://mp-prod.appsmb.com/mas/v5/app/proxy?alias=', 'sseUrl': 'https://sse.msmartlife.net', 'idAdapt': '131329', 'countryCode': 'HK'}
DEBUG:midea_beautiful.cloud:HTTP request /v1/user/login/id/get: {'x-recipe-app': '1010', 'Authorization': 'Basic <redacted>=', 'sign': '<redacted>', 'secretVersion': '1', 'random': '1691159234', 'version': '2.22.0', 'systemVersion': '8.1.0', 'platform': '0', 'Accept-Encoding': 'identity', 'Content-Type': 'application/json'} {'appId': 1010, 'format': 2, 'clientType': 1, 'language': 'en_US', 'src': 1010, 'stamp': '20230804102714', 'loginAccount': 'david@<redacted>', 'appVNum': '2.22.0', 'appVersion': '2.22.0', 'clientVersion': '2.22.0', 'platformId': '1', 'reqId': 'acc4fd5448c0a9bf8c4be0f5cfcd8b94', 'retryCount': '3', 'uid': '', 'userType': '0'}
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): mp-prod.appsmb.com:443
DEBUG:urllib3.connectionpool:https://mp-prod.appsmb.com:443 "POST /mas/v5/app/proxy?alias=/v1/user/login/id/get HTTP/1.1" 200 41
DEBUG:midea_beautiful.cloud:HTTP response text: {"code":"3004","msg":"value is illegal."}
DEBUG:midea_beautiful.cloud:HTTP response: {'code': '3004', 'msg': 'value is illegal.'}
DEBUG:midea_beautiful.cloud:Restarting session: '3004' - 'value is illegal.'
```
Like @Enton-72 I am also willing to help debug.

**Comment by dkerr64** [comment]
An update that I was successful using the NetHome Plus app.  First I downloaded and installed that app on my iPhone, then registered for a new account with it (I used the same email address that I used with the mSmartHome app) and then connected the dehumidifier to it.  Now I was able to retrieve the credentials...

```
root@homebridge:~# midea-beautiful-air-cli discover --address 192.168.29.131 --account David@<redacted> --password <redacted> --credentials
id 000000P0000000Q1B88C291B363A0000/150633093209019
  id      = 150633093209019
  addr    = 192.168.29.131
  s/n     = 000000P0000000Q1B88C291B363A0000
  model   = Dehumidifier
  ssid    = net_a1_363A
  online  = True
  name    = Dehumidifier_363A
  running = True
  humid%  = 50
  target% = 40
  temp    = 23.0
  fan     = 80
  tank    = False
  mode    = 1
  ion     = False
  filter  = False
  pump    = False
  defrost = False
  sleep   = False
  error   = 0
  supports= {'fan_speed': 3, 'filter': 2, 'dry_clothes': 1, 'water_level': 4}
  version = 3
  token   = 9B1403CC...<redacted>...2311C5743
  key     = 30dc62ba...<redacted>...97dcc9
```
and....
```
root@homebridge:~# midea-beautiful-air-cli status --ip 192.168.29.131 --token 9B1403CC...<redacted>...2311C5743 --key 30dc62ba...<redacted>...97dcc9
id 000000P0000000Q1B88C291B363A0000/150633093209019
  id      = 150633093209019
  addr    = 192.168.29.131
  s/n     = 000000P0000000Q1B88C291B363A0000
  model   = Dehumidifier
  ssid    = net_a1_363A
  online  = True
  name    = 150633093209019
  running = True
  humid%  = 51
  target% = 40
  temp    = 23.0
  fan     = 80
  tank    = False
  mode    = 1
  ion     = False
  filter  = False
  pump    = False
  defrost = False
  sleep   = False
  error   = 0
  supports= {'fan_speed': 3, 'filter': 2, 'dry_clothes': 1, 'water_level': 4}
  version = 3
```

On my iPhone the device is visible and can be controlled in both apps.  I do not know if this will cause future problems or no.t

**Comment by dereitz** [comment]
> An update that I was successful using the NetHome Plus app. First I downloaded and installed that app on my iPhone, then registered for a new account with it (I used the same email address that I used with the mSmartHome app) and then connected the dehumidifier to it. Now I was able to retrieve the credentials...
> 
> ```
> root@homebridge:~# midea-beautiful-air-cli discover --address 192.168.29.131 --account David@<redacted> --password <redacted> --credentials
> id 000000P0000000Q1B88C291B363A0000/150633093209019
>   id      = 150633093209019
>   addr    = 192.168.29.131
>   s/n     = 000000P0000000Q1B88C291B363A0000
>   model   = Dehumidifier
>   ssid    = net_a1_363A
>   online  = True
>   name    = Dehumidifier_363A
>   running = True
>   humid%  = 50
>   target% = 40
>   temp    = 23.0
>   fan     = 80
>   tank    = False
>   mode    = 1
>   ion     = False
>   filter  = False
>   pump    = False
>   defrost = False
>   sleep   = False
>   error   = 0
>   supports= {'fan_speed': 3, 'filter': 2, 'dry_clothes': 1, 'water_level': 4}
>   version = 3
>   token   = 9B1403CC...<redacted>...2311C5743
>   key     = 30dc62ba...<redacted>...97dcc9
> ```
> 
> and....
> 
> ```
> root@homebridge:~# midea-beautiful-air-cli status --ip 192.168.29.131 --token 9B1403CC...<redacted>...2311C5743 --key 30dc62ba...<redacted>...97dcc9
> id 000000P0000000Q1B88C291B363A0000/150633093209019
>   id      = 150633093209019
>   addr    = 192.168.29.131
>   s/n     = 000000P0000000Q1B88C291B363A0000
>   model   = Dehumidifier
>   ssid    = net_a1_363A
>   online  = True
>   name    = 150633093209019
>   running = True
>   humid%  = 51
>   target% = 40
>   temp    = 23.0
>   fan     = 80
>   tank    = False
>   mode    = 1
>   ion     = False
>   filter  = False
>   pump    = False
>   defrost = False
>   sleep   = False
>   error   = 0
>   supports= {'fan_speed': 3, 'filter': 2, 'dry_clothes': 1, 'water_level': 4}
>   version = 3
> ```
> 
> On my iPhone the device is visible and can be controlled in both apps. I do not know if this will cause future problems or not.

Thank you for the tip!  This worked for me!

**Comment by cdickmann** [comment]
I have the same original issue, haven't tried the NetHome app yet.

**Comment by nbogojevic** [comment]
Should be fixed in version 0.10.0

**Comment by sjhhagl** [comment]
I'm seeing what seems to be the same issue with version 0.10.5 with two humidifiers registered in the MSmartHome app.


>$ midea-beautiful-air-cli discover --address 192.168.2.86
id 000000P0000000Q1305223AA40A60000/151732605659762
  id      = 151732605659762
  addr    = 192.168.2.86
  s/n     = 000000P0000000Q1305223AA40A60000
  model   = Dehumidifier
  ssid    = net_a1_40A6
  online  = True
  name    = 151732605659762
  running = False
  humid%  = 45
  target% = 50
  temp    = 0
  fan     = 40
  tank    = False
  mode    = 0
  ion     = False
  filter  = False
  pump    = False
  defrost = False
  sleep   = False
  error   = 0 
  supports= {}
  version = 3

Then I point it at their cloud:
>$ midea-beautiful-air-cli discover --address 192.168.2.686--credentials --account REDACTED --password REDACTED --app MSmartHome 
WARNING:midea_beautiful.scanner:Some appliance(s) where not discovered on local network: 0 discovered out of 2
WARNING:midea_beautiful.scanner:Unable to discover registered appliance {'id': '15063309403****', 'name': 'Basement', 'sn': '000000P0000000Q1305223A3********', 'type': '0xA1', 'modelNumber': '41377'}
WARNING:midea_beautiful.scanner:Unable to discover registered appliance {'id': '15173260565****', 'name': 'Kitchen', 'sn': '000000P0000000Q1305223AA********', 'type': '0xA1', 'modelNumber': '41377'}

When run with --verbose --log DEBUG I can see that it's talking to the device and getting a reply, but failing with "value is illegal" when requesting the token.

>DEBUG:midea_beautiful.scanner:UDP broadcast 192.168.2.86:20086 b'ZZ\x01\x11H\x00\x92\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x7fu\xbdk>O\x8bv.\x84\x9cnW\x8de\x90\x03n\x9dCB\xa5\x0f\x1fV\x9e\xb8\xec\x91\x8e\x92\xe5'
DEBUG:midea_beautiful.scanner:Reply from address=192.168.2.86 payload=b'\x83p\x00\xc8 \x0f\x00\x00ZZ\x01\x11\xb8\x00z\x80\x00\x00\x00\x00\x18\x0e3\x14\x02\x04\x19\x14r\xaa\x0f\x00\x00\x8a\x00\x00\x00\x00\x00\x00\x00\x00\x01\x80\x00\x00\x00\x00P\xd5\xbf\xb8\xd4\x9e\xc3\x08\x14\x99L\x16\xbb\xb6\x8ef\xa7\x9d\x07\xce\xee\xef\xcc\x18LB\x83\xf92w\xcb\x9a\n\xf3g\xb54\x1b\xf6y\xcbh\xea\xb8\x1c\x1b\xfc\x0em\xdf\xade\xde\xc2Q\x80\xed\xd5kEQ\x8b\x8c\xf6%3\xabq\x05\xd9\xc8/\x9f\xb8\x1b\xcf\xf8X\xba\xdb\xf6+\x1d\x8b\xb9\xe5\xc2]\xba{\xf8\xe0\xccLw\x94K\xdf\xb3\xe1n3\xd8\x87h\xccL=\x06X\x93}\x0b\xb1\x93i\xbf\x03\x17\xb2M:M\xe9\xe6\xa11\x06S\x14q\x0fi+\xa0l\x1e\x05Q`\xc2\n@\xb5rt\x8f?\x19n\xebS\xadX\xefp\xea\xfd\xd00'
DEBUG:midea_beautiful.lan:From 192.168***** decrypted reply=b'V\x02\xa8\xc0,\x19\x00\x00000000P0000000Q1305223AA********\x0bnet_a1_40A6\x00\x00\x87\x00\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\xa1\x00\xa1\xa1\x00\x00\x00\x000R#\xaa@\xa6\x15\x00#\x08!"\x00\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
DEBUG:midea_beautiful.appliance:Creating DehumidifierAppliance 15173260565****
DEBUG:midea_beautiful.lan:Descriptor data '{id=15173260565****, address=192.168*****, port=6444, version=3, name=15173260565****, online=True, type=0xa1, subtype=a1a1, flags=0, extra=2, reserved=87, mac=305223a*****, ssid=net_a1_40A6, udp_version=0, protocol=150023, version=8.33.34, sn=000000P0000000Q1305223AA********, state=[Dehumidifier]{id=15173260565****, type=0xa1, mode=0, running=False, target_humidity=50, fan_speed=40, tank_full=False current_humidity=45, current_temperature=0 defrosting=False, filter=False, tank_level=0, error_code=0, prompt=False, supports={}}}'
DEBUG:midea_beautiful.scanner:Finished broadcast collection
DEBUG:midea_beautiful.cloud:HTTP request /v1/iot/secure/getToken: {'uid': '********************************', 'accessToken': '****************', 'x-recipe-app': '1010', 'Authorization': 'Basic ********************************************************', 'sign': '6fdac24f5d33bba95d5def8dd531695324f981ced45e09f494abcba6d5d90332', 'secretVersion': '1', 'random': '1743627077', 'version': '2.22.0', 'systemVersion': '8.1.0', 'platform': '0', 'Accept-Encoding': 'identity', 'Content-Type': 'application/json'} {'appId': 1010, 'format': 2, 'clientType': 1, 'language': 'en_US', 'src': 1010, 'stamp': '20250402145117', 'deviceId': 'c1acad8939ac0d7d', 'udpid': 'ab5f59c5bc30aaf6d5bff1d67c4100ea', 'appVNum': '2.22.0', 'appVersion': '2.22.0', 'clientVersion': '2.22.0', 'platformId': '1', 'reqId': 'b9c09d0a8b984715b126d42a0ecc95e5', 'retryCount': '3', 'uid': '********************************', 'userType': '0'}
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): mp-prod.appsmb.com:443
DEBUG:urllib3.connectionpool:https://mp-prod.appsmb.com:443 "POST /mas/v5/app/proxy?alias=/v1/iot/secure/getToken HTTP/1.1" 200 41
DEBUG:midea_beautiful.cloud:HTTP response: *** REDACTED ***
DEBUG:midea_beautiful.cloud:Restarting session: '3004' - 'value is illegal.'
